secrets/secrets.go

```diff
@@ -0,0 +1,172 @@
+package secrets
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"sync"
+)
+
+var (
+	ErrSecretsNotFound = errors.New("Secrets Implementor not registerd")
+	secretKeyOkMsg     = "Secret Key set successfully"
+	secretLoginOkMsg   = "Login completd"
+)
+
+type SecretManager interface {
+	// Login create session with secret store
+	SecretLogin(secretType int, secretConfig map[string]string) error
+	// SetClusterSecretKey sets the cluster wide secret key
+	SetClusterSecretKey(secretKey string, override bool) error
+	//CheckSecretLogin validates session with secret store
+	CheckSecretLogin() error
+	//SetSecret sets secret key against data
+	SetSecret(key string, value string) error
+	// GetSecret retrieves the data for the given feature and key
+	GetSecret(key string) (string, error)
+	//Routes to the secret API
+	Routes() []*Route
```

Comment:
Remove. Routes are specific to a REST service. A SecretManager is an implementation of an interface which does not matter how data gets to it. For example, another day we could change to gRPC or whatever method of communication, but the SecretManager would be the same interface implementation.